progress-bar.js

```diff
@@ -0,0 +1,93 @@
+(function(){
+    if (!document.addEventListener ||
+        (  document.documentElement.style.transform === undefined
+        && document.documentElement.style.webkitTransform === undefined)){
+        return;
+    }
+
+    var TRANSFORM = document.documentElement.style.webkitTransform !== undefined ? 'webkitTransform' : 'transform';
+
+    var options = INSTALL_OPTIONS;
+
+    var containerEl = null;
+    var barEl = null;
+    var render = function(){
+        if (!document.body)
+            return;
+
+        if (!containerEl){
+            containerEl = document.createElement('div');
+            document.body.appendChild(containerEl);
+
+            barEl = document.createElement('div');
+            barEl.className = 'scroll-progress-bar-bar';
+            containerEl.appendChild(barEl);
+        }
+
+        containerEl.className = 'scroll-progress-bar-container';
+        containerEl.className += ' scroll-progress-bar-' + options.position;
+
+        barEl.style.height = options.thickness + "px";
+        barEl.style.backgroundColor = options.barColor;
+
+        if (options.containerHasColor)
+            containerEl.style.backgroundColor = options.containerColor;
+        else
+            containerEl.style.backgroundColor = '';
+
+        updateProgress();
+    };
+
+    var now = function(){
+        return (window.performance && performance.now()) || (+ new Date);
+    };
+
+    var lastCall = 0;
+    var updateProgress = function(){
+        if (!barEl)
+            return;
+
+        // Don't allow updates faster than 60 FPS
+        if (now() - lastCall < 15)
+            return;
+
+        lastCall = now();
+
+        var top = window.scrollY;
+        var height = document.body.getBoundingClientRect().height - window.innerHeight;
+
+        var percent = Math.min(top / height * 100, 100);
```

Comment:
If you use `Math.ceil` to round here, the bar ends up being a bit janky, as the pixel values are being rounded off.  There is no need to round the value though, as the CSS is more than happy to accept a fractional value.